src/main/java/tc/oc/pgm/restart/RestartManager.java

```diff
@@ -94,6 +96,28 @@ public boolean isRestartDeferredBy(RequestRestartEvent.Deferral deferral) {
     return this.deferrals.contains(deferral);
   }
 
+  public void requestTimedRestart(Duration duration) {
+    Instant queueStamp = Instant.now();
+    long ticks = duration.getStandardSeconds() * 20;
+    timedQueuedRestartTaskID =
+        this.plugin
+            .getServer()
+            .getScheduler()
+            .runTaskLater(
+                this.plugin,
+                () -> {
+                  if (!this.deferrals.isEmpty()) {
+                    timedQueuedRestartTaskID = 0;
+                    return;
+                  }
+                  timedQueuedRestartTaskID = 0;
```

Comment:
If you're going to reset the taskId anyways, just place it once before the conditional